### PR TITLE
mixin: Added critical Rules alerts.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
BTW all those alerts assume you have some rules, but I guess that's fair.

File: mixin/thanos/alerts/rule.libsonnet [hunk 1]
```diff
@@ -153,6 +153,33 @@
               severity: 'warning',
             },
           },
+          {
+            alert: 'ThanosRuleNoEvaluationForLast10Intervals',
+            annotations: {
+              message: 'Thanos Rule {{$labels.job}} have {{ $value | humanize }}% rule groups that did not evaluate for at least 10x of their expected interval.',
```
Comment (metalmatze) [inline mixin/thanos/alerts/rule.libsonnet:159]:
```suggestion
              message: 'Thanos Rule {{$labels.job}} has {{ $value | humanize }}% rule groups that did not evaluate for at least 10x of their expected interval.',
```

File: mixin/thanos/alerts/rule.libsonnet [hunk 2]
```diff
@@ -153,6 +153,33 @@
               severity: 'warning',
             },
           },
+          {
+            alert: 'ThanosRuleNoEvaluationForLast10Intervals',
```
Comment (metalmatze) [inline mixin/thanos/alerts/rule.libsonnet:157]: This is more of a nitpick or maybe even opinion already: 

```suggestion
            alert: 'ThanosRuleNoEvaluationFor10Intervals',
```

File: mixin/thanos/alerts/rule.libsonnet [hunk 3]
```diff
@@ -153,6 +153,33 @@
...
+            annotations: {
+              message: 'Thanos Rule {{$labels.job}} have {{ $value | humanize }}% rule groups that did not evaluate for at least 10x of their expected interval.',
+            },
+            expr: |||
+              time() - prometheus_rule_group_last_evaluation_timestamp_seconds{%(selector)s}
+                  >  10 * prometheus_rule_group_interval_seconds{%(selector)s}
```
Comment (metalmatze) [inline mixin/thanos/alerts/rule.libsonnet:163]: Some spacing might be nice here to more quickly see what's happening.
```suggestion
              time() - prometheus_rule_group_last_evaluation_timestamp_seconds{%(selector)s}
              >
              10 * prometheus_rule_group_interval_seconds{%(selector)s}
```

File: mixin/thanos/alerts/rule.libsonnet (hunk 1 above)
Comment (bwplotka) [reply mixin/thanos/alerts/rule.libsonnet:159]: dude, you have `have` everywhere else (: Will fix there as well

File: mixin/thanos/alerts/rule.libsonnet [hunk 4]
```diff
@@ -153,6 +153,33 @@
...
+            ||| % thanos.rule,
+            'for': '5m',
+            labels: {
+              severity: 'critical',
+            },
+          },
+          {
+            alert: 'ThanosRuleTSDBNotIngestingSamples',
+            annotations: {
+              message: 'Thanos Rule {{$labels.job}} did not ingested any samples for last 15 minutes.',
```
Comment (metalmatze) [inline mixin/thanos/alerts/rule.libsonnet:173]:
```suggestion
              message: 'Thanos Rule {{$labels.job}} did not ingest any samples for the last 15 minutes.',
```

File: mixin/thanos/alerts/rule.libsonnet (hunk 3 above)
Comment (bwplotka) [reply mixin/thanos/alerts/rule.libsonnet:163]: we need PromQL fmt (:

File: mixin/thanos/alerts/rule.libsonnet (hunk 1 above)
Comment (metalmatze) [reply mixin/thanos/alerts/rule.libsonnet:159]: Well, then that's wrong. Also, I didn't write most of the other ones. Let's not start blaming here anyway :)

**Comment by bwplotka** [comment]
Addressed all @metalmatze

**Pull request merged**